### PR TITLE
simplify internal locations API

### PR DESCRIPTION
**Pull request by sqs**
See commits. This just cleans up some code that I had written a while back and adds some tests.

Prereq for #1697

### CODE REVIEW

**Comment by felixfbecker** [comment]
Great refactor, but I see merge conflict hell incoming with my comlink branch 😭

**Comment by sqs** [comment]
Cool. I’ll hold off.

File: shared/src/api/client/services/location.ts [hunk 1]
```diff
@@ -57,55 +61,6 @@ export class TextDocumentLocationProviderRegistry<
...
 }
 
-/**
- * Returns an observable that emits the providers' location results whenever any of the last-emitted set of
- * providers emits hovers.
- *
- * Most callers should use the TextDocumentLocationProviderRegistry class, which uses the registered providers
- * (instead of this function, which requires the caller to specify the providers to get locations from).
- */
-export function getLocations<
```
Comment (sqs) [inline shared/src/api/client/services/location.ts]: This func is just moved down below and renamed to getLocationsFromProviders for clarity.

**Comment by sqs** [comment]
@felixfbecker 	@lguychard 	PTAL, updated. This is just a refactor. It ended up having basically no merge conflicts with Comlink :tada:.

**Pull request merged**